Examples/Sources/MacroExamples/Implementation/Extension/ProtocolDefaultImplMacro.swift

```diff
@@ -0,0 +1,50 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+import SwiftSyntax
+import SwiftSyntaxMacros
+import SwiftSyntaxBuilder
+
+public enum ProtocolDefaultImplMacro: ExtensionMacro {
+  public static func expansion(
+    of node: AttributeSyntax,
+    attachedTo declaration: some DeclGroupSyntax,
+    providingExtensionsOf type: some TypeSyntaxProtocol,
+    conformingTo protocols: [TypeSyntax],
+    in context: some MacroExpansionContext
+  ) throws -> [ExtensionDeclSyntax] {
+    // Ensure that the declaration is a protocol declaration
+    guard let protocolDecl = declaration.as(ProtocolDeclSyntax.self) else {
+      return []
+    }
+
+    // Retrieve all methods from the protocol
+    var methods = protocolDecl.memberBlock.members
+      .map(\.decl)
+      .compactMap { $0.as(FunctionDeclSyntax.self) }
+
+    // Add implementation to all methods from the protocol
+    methods.indices.forEach { index in
+      methods[index].body = CodeBlockSyntax {
+        ExprSyntax(#"fatalError("whoops 😅")"#)
+      }
+    }
```

Comment:
You should be able to simplify this further. Also, usage of `forEach` is generally considered non-idiomatic Swift.

```suggestion
    var methods = protocolDecl.memberBlock.members
      .map(\.decl)
      .compactMap { 
        guard var function = $0.as(FunctionDeclSyntax.self) else {
          return nil
        }
        function.body = CodeBlockSyntax {
          ExprSyntax(#"fatalError("whoops 😅")"#)
        }
        return function
      }
```

Reply:
Oh, that's much simpler—done 😅